### DISCUSSION TITLE
pre-proposal: Notary Transaction

### DISCUSSION THREAD

**Discussion by rothexD** [XRPLF/XRPL-Standards#193] [XLS Ideas (pre standard proposal)]
-------------
Title: Notary Transaction
Revision:     1 (2024-04-14)

Authors:   
  rothexD

Affiliation:
  Broke student

--------
# 1. Introduction

Currently assets have to be transferd and held by a third party that can use these assets as they wish. This can lead to potentiall missuse. This is unavoidable in real life but can be avoided using blockchains. With this type of transactions the assets can be managed by the third party but not missused. 

## 1.1 Use-Case

1. Currently in many juristictions assets have to be held by a notary until certain things have happend. For example a transfer of assets may have been agreed on by 2 parties but until the documentation has went through all goverment processes the assets have to be held by a notary or similar.
2. A digital shop may facilitate transactions between customers and businesses. The funds will be in the shops hand until the items have been shipped/refund period is over etc.. The shop can run with the funds/assets as they are holding it directly

## Terminology 
`Issuer`: the person starting the transfer of assets
`Notary`: the person managing the state of the transaction
`Receiver`: the person receiving the assets

# 2. Design Thoughts

1. The Issuer starts a `Notary Transaction`  with an EndDateTime 
2. The `Notary Transaction` locks the assets until the EndDatetTime or an action was performed by the `Notary`
3. The `Notary Transaction` can include a `Notary Fee` either fixed, incase of a NFT/Currency or % incase of Currency
4. The Issuer can set a `ReleaseFlag` for who the funds should go to if the `Notary` does not release the Assets until enddatetime
5.1 Either the funds go back to the `Issuer` or `Receiver`
6. The `Notary` can release/partially complete/cancel the transfer of assets until the enddatetime to the `Receiver`
7. For release and cancel `Notary` automatically gets the full amount, for partial the `Notary` gets a partial amount of the `Notary Fee`
8. The `Issuer` and the `Receiver` can release the Assets past its EndDateTime following the `ReleaseFlag` logic

# 3. Limits
Currently i did not think about assets that are frozen or have clawback enabled

_This is currently very early on and no idea if this would be the best way of doing it. I am working on writing this up 🤠_

**Comment by mvadari** [comment]
I recommend writing this as a pre-proposal idea since it seems like the details are still being worked out, and publishing an XLS once you have a full write-up :)

**Reply by rothexD** [reply]
Thx I thought I published it as a preprosal will update

**Reply by mvadari** [reply]
Oh, you did, sorry - pre-proposals don't get XLS numbers, so just remove that part 🙂 You can claim an XLS number once you're ready.

**Comment by sappenin** [comment]
Hi @rothexD - thanks for starting your discussion. 

I'm curious - could the `notary` role be performed by the ledger itself using the [Escrow](https://xrpl.org/docs/references/protocol/ledger-data/ledger-entry-types/escrow/#escrow) feature? 

Or perhaps instead: are you looking for a few 3rd parties to coordinate some assets in a single account? In that case, I wonder if [multisig](https://xrpl.org/docs/concepts/accounts/multi-signing/) could be used. E.g., have a single account that requires (e.g.) a 2-of-2 multi-sign to move funds (e.g., a sig from the `notary` and another one from the `docs provider` that you allude to in your description).

(I'm trying to determine if what you're proposing is already possible with existing functionality)

**Reply by rothexD** [reply]
I had another look at `Escrow` and  played with my different scenarios it's a bit complicated to setup but i think it would be possible using `Escrow`

Thanks a lot @sappenin
